doc/connectivity/networking/net_config_guide.rst

```diff
@@ -0,0 +1,131 @@
+   "Maximum number of TLS/DTLS ciphersuites per socket.
+   This variable sets maximum number of TLS/DTLS ciphersuites that can
+   be used with specific socket, if set explicitly by socket option.
+   By default, all ciphersuites that are available in the system are
+   available to the socket."
+   ":kconfig:option:`CONFIG_NET_SOCKETS_TLS_MAX_APP_PROTOCOLS`",
+   "Maximum number of supported application layer protocols"
+   This variable sets maximum number of supported application layer
+   protocols over TLS/DTLS that can be set explicitly by a socket option.
+   By default, no supported application layer protocol is set."
+   ":kconfig:option:`CONFIG_NET_SOCKETS_TLS_MAX_CLIENT_SESSION_COUNT`",
+   "This variable specifies maximum number of stored TLS/DTLS sessions,
+   used for TLS/DTLS session resumption."
+   ":kconfig:option:`CONFIG_NET_SOCKETPAIR_BUFFER_SIZE`",
+   "This option is used by socketpair() function. It sets the size of the
+   internal intermediate buffer, in bytes. This sets the limit how large
+   messages can be passed between two socketpair endpoints."
+   ":kconfig:option:`CONFIG_NET_TCP_TIME_WAIT_DELAY`",
+   "How long to wait in TCP TIME_WAIT state (in milliseconds).
+   To avoid a (low-probability) issue when delayed packets from
+   previous connection get delivered to next connection reusing
+   the same local/remote ports, RFC 793 (TCP) suggests to keep
+   an old, closed connection in a special "TIME_WAIT" state for
+   the duration of 2*MSL (Maximum Segment Lifetime). The RFC
+   suggests to use MSL of 2 minutes, but notes "This is an
+   engineering choice, and may be changed if experience indicates
+   it is desirable to do so." For low-resource systems, having
+   large MSL may lead to quick resource exhaustion (and related
+   DoS attacks). At the same time, the issue of packet misdelivery
+   is largely alleviated in the modern TCP stacks by using random,
+   non-repeating port numbers and initial sequence numbers. Due
+   to this, Zephyr uses much lower value of 1500ms by default.
+   Value of 0 disables TIME_WAIT state completely."
+   ":kconfig:option:`CONFIG_NET_TCP_RETRY_COUNT`",
+   "Maximum number of TCP segment retransmissions.
+   The following formula can be used to determine the time (in ms)
+   that a segment will be be buffered awaiting retransmission:
+   n=NET_TCP_RETRY_COUNT
+   Sum((1<<n) * NET_TCP_INIT_RETRANSMISSION_TIMEOUT)
+   n=0
+   With the default value of 9, the IP stack will try to
+   retransmit for up to 1:42 minutes.  This is as close as possible
+   to the minimum value recommended by RFC1122 (1:40 minutes).
+   Only 5 bits are dedicated for the retransmission count, so accepted
+   values are in the 0-31 range.  It's highly recommended to not go
+   below 9, though.
+   Should a retransmission timeout occur, the receive callback is
+   called with -ECONNRESET error code and the context is dereferenced."
+   ":kconfig:option:`CONFIG_NET_TCP_MAX_SEND_WINDOW_SIZE`",
```

Comment:
I think we should elaborate here on the impact of the window size on the net_buf usage. I. e. that this value might require finetuning (lowering) when it is expected to have multiple active TCP connections in the system, otherwise multiple TCP connections could easily exhaust net_buf pool for the queued data.